frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts

```diff
@@ -159,9 +159,67 @@ const osUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
     vmWizardInternalActions[InternalActionType.UpdateVmSettingsField](
       id,
       VMSettingsField.MOUNT_WINDOWS_GUEST_TOOLS,
-      { isHidden: asHidden(!isWindows, VMSettingsField.OPERATING_SYSTEM), value: isWindows },
+      {
+        isHidden: asHidden(!isWindows, VMSettingsField.OPERATING_SYSTEM),
+        value: isWindows,
+      },
     ),
   );
+
+  const relevantOptions = iGetRelevantTemplateSelectors(state, id);
+  const iCommonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+  const iTemplate =
+    iCommonTemplates && iGetRelevantTemplate(null, iCommonTemplates, relevantOptions);
+
+  if (os && !iTemplate) {
+    const keepFlavor = Object.assign({}, relevantOptions);
+    Object.assign(keepFlavor, { workload: null });
+    // Check for a valid common-template that contain the chosen Flavor
```

Comment:
let's not try to find combinations which could work. We don't know if the user prefers to keep his flavor or workload so let's reset both if we have to reset one of them.